### PR TITLE
fix: Validate docker volume names when creating new project.

### PR DESCRIPTION
**Pull request by Isakdl**
# Changes

- Checks that the docker volume does not exist, if it does give an error to the user.
- In maintenance mode only try to reconnect to the database 6 times before giving up.

Closes: #531

## Pre-launch Checklist

- [ ] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [ ] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [ ] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [ ] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

_If you have done any breaking changes, make sure to outline them here, so that they can be included in the notes for the next release._

### CODE REVIEW

File: tools/serverpod_cli/lib/src/create/create.dart [hunk 1]
```diff
@@ -96,7 +95,7 @@ Future<bool> performCreate(
         'also install and configure Postgres and Redis manually and run this '
         'command with the -f flag added.';
     var strIssueDockerVolume =
-        'A docker volume with the name "${SetupConstants.dockerVolumeName(name)}" already exists. '
+        'A docker volume with the nameg "${SetupConstants.dockerVolumeName(name)}" already exists. '
```
Comment (SandPod) [inline tools/serverpod_cli/lib/src/create/create.dart:98]: chore: Change `nameg` to `name`.

**Review by SandPod — COMMENTED** [review]
Nice 👍 

I had two questions that I would like answers on before an approve, and a minor fix.

File: packages/serverpod/lib/src/server/serverpod.dart [hunk 2]
```diff
@@ -632,6 +641,13 @@ class Serverpod {
         }
 
         await session.close();
+
+        if (maxAttempts != null && attempts >= maxAttempts) {
+          throw TimeoutException(
```
[the remark this answers is not in the document]
Comment (Isakdl) [reply packages/serverpod/lib/src/server/serverpod.dart:646]: We do not, but we also only throw this in maintenance mode currently.

File: tools/serverpod_cli/lib/src/util/constants.dart [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class SetupConstants {
+  static String dockerVolumeName(String projectName) =>
+      '${projectName}_server_${projectName}_data';
```
[the remark this answers is not in the document]
Comment (Isakdl) [reply tools/serverpod_cli/lib/src/util/constants.dart:3]: The problem here would be that if this happens we will fail "silently" I.E. how we fail today.

[the remark this answers is not in the document]
Comment (SandPod) [reply tools/serverpod_cli/lib/src/util/constants.dart:3]: True, but we would fall all the time in the released version of Serverpod. Might help someone that doesn't have this connection knowledge solve it 🤷

[the remark this answers is not in the document]
Comment (SandPod) [reply tools/serverpod_cli/lib/src/util/constants.dart:3]: If none of us are looking at this :)

[the remark this answers is not in the document]
Comment (Isakdl) [reply tools/serverpod_cli/lib/src/util/constants.dart:3]: I think a comment can help us maybe, but not a user of Serverpod.

**Pull request merged**